src/testing/testing_buildBlockV1.md

```diff
@@ -0,0 +1,115 @@
+# `testing_buildBlockV1`
+
+This document specifies the `testing_buildBlockV1` RPC method. This method is a debugging and testing tool that simplifies the block production process into a single call. It is intended to replace the multi-step workflow of sending transactions, calling `engine_forkchoiceUpdated` with `payloadAttributes`, and then calling `engine_getPayload`.
+
+This method is considered sensitive and is intended for testing environments only. See [**Security Considerations**](#security-considerations) for more details.
+
+### Request
+
+* method: `testing_buildBlockV1`
+* params:
+  1. `parentBlockHash`: `DATA`, 32 Bytes - block hash of the parent of the requested block.
+  2. `payloadAttributes`: `Object` - instance of [`PayloadAttributesV3`](../engine/cancun.md#payloadattributesv3)
+  3. `transactions`: `Array of DATA` - an array of raw, signed transactions (hex-encoded `0x...` strings) to forcibly include in the generated block.
+  5. `extraData`: `DATA|null`, 0 to 32 Bytes - data to be set as the `extraData` field of the built block.
+
+### Response
+
+* result: `OBJECT` - instance of [`engine_getPayloadV4`](../engine/prague.md#response-1)
+* error: code and message set in case an exception happens while building the requested block.
+
+### Specification
+
+* The client MUST build a new execution payload using the block specified by `parentBlockHash` as its parent.
+
+* The client MUST use the provided `payloadAttributes` to define the context of the new block.
+
+* The client MUST include all transactions from the `transactions` array on the block's transaction list, in the order they were provided.
+
+* The client MUST NOT include any transactions from its local transaction pool. The resulting block MUST only contain the transactions specified in the `transactions` array.
```

Comment:
```suggestion
* If the `transactions` array is non-empty, the client MUST NOT include any transactions from its local transaction pool. The resulting block MUST only contain the transactions specified in the `transactions` array.
```

It might be a useful (optional?) feature to allow for selectiung txpool transactions in the case where the transaction array is empty.